design-proposals/primary_network_binding_for_seamless_migration.md

```diff
@@ -0,0 +1,459 @@
+commonly used, it may add redundant overhead.
+As a plugin, it can be developed independently to the Kubevirt release
+cycle on a different repo.
+
+#### Macvtap Binding
+
+```mermaid
+graph LR;
+    subgraph virt-launcher
+    VM <--> macvtap0
+    macvtap0 <--> eth0
+    end
+    subgraph Node
+    eth0 <--> veth123
+    end
+```
+
+The macvtap binding is an alternative to the bridge binding with very
+similar pros and cons.
+The [macvtap device](https://virt.kernelnewbies.org/MacVTap) is attached
+to the pod interface in a passthru mode.
+
+> *Warning*: Requires PoC to confirm it works when attaching to a veth
+> interface and multiple mac addresses are active in the guest.
+
+Deployed as a binding plugin with no option to self-serve IPAM to the VM.
+It is dependent on the network-provider to be able and serve IP
+addresses to the guest.
+
+Compared to the bridge binding (as a plugin), it is simpler and with
+probably better performance.
+In addition, macvtap devices have a device representation and can be
+relabeled (SELinux) post creation.
+
+> *Warning*: Relabeling of a macvtap device needs PoC confirmation.
+
+As a plugin, it can be developed independently to the Kubevirt release
+cycle on a different repo.
+
+#### Other L2 Bindings
+
+The bridge and macvtap bindings have one behavior in common: They
+forward traffic based on L2 (ethernet frames) information only.
+
+Similar bindings may be invented to forward traffic using other technologies.
+Some may be more efficient than others with different complexities but
+with a very similar pros and cons as seen on the previous bridge &
+macvtap bindings, e.g.:
+
+- Passing the IP & route configuration from the pod interface to the VM
+  guest interface is left to the network provider (e.g. DHCPv4, DHCPv6, RA).
+
+Some of the suggested L2 alternatives are:
+
+- OVS bridge
+- eBPF
+- Tap device directly connected to the node, passed into the pod for
+  consumption (saving the veth pair that links the pod interface to the
+  node).
+
+#### Summary
+
+| Feature\\Binding            | Passt                                                                                                           | Bridge[*]                            | Macvtap                                          |
+|:----------------------------|:----------------------------------------------------------------------------------------------------------------|:-------------------------------------|:-------------------------------------------------|
+| Deployment                  | Binding Plugin                                                                                                  | Core                                 | Binding Plugin                                   |
+| IP & Route delivery         | Self (IPv4 & IPV6)                                                                                              | Self for IPv4, Net Provider for IPv6 | Net Provider                                     |
```

Comment:
We should add here another row related to VM probes, I think with passt we support them with the bridge and macvtap we do not, the ovn-k version of primary network the probes traffic goes to eth0, same as kubevirt not supporting probes for secondary interfaces.